File: P1 2016.2/Questao(2)/python/entities/produtos/Liquidificador.py
from .Produto import Produto
from .Ligavel import Ligavel


class Liquidificador(Produto, Ligavel):
    def __init__(self, preco: float, id: int, marca: str) -> None:
        super().__init__(preco, id)
        self.marca = marca

    def ligar(self):
        print("Ligando o liquidificador da marca:", self.marca)

    def __str__(self) -> str:
        return f"Preco: {self.preco}, ID:{self.id}, Marca:{self.marca}"

File: P1 2016.2/Questao(2)/java/entities/produtos/Liquidificador.java
package entities.produtos;

public class Liquidificador extends Produto implements Ligavel {
    public String marca;

    public Liquidificador(double preco, int id, String marca) {
        super(preco, id);
        this.marca = marca;
    }

    public String getMarca() {
        return marca;
    }

    public void ligar() {
        System.out.println("Ligando o liquidificador");
    }
}
